File: google-common/src/test/java/docs/javadsl/GoogleCommonDoc.java
/*
 * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
 */

package docs.javadsl;

import akka.actor.ActorSystem;
import akka.stream.Graph;
import akka.stream.alpakka.google.GoogleAttributes;
import akka.stream.alpakka.google.GoogleSettings;
import akka.stream.javadsl.Source;

public class GoogleCommonDoc {

  ActorSystem system = null;
  Graph<?, ?> stream = null;

  void accessingSettings() {
    // #accessing-settings
    GoogleSettings defaultSettings = GoogleSettings.create(system);
    GoogleSettings customSettings = GoogleSettings.create("my-app.custom-google-config", system);
    Source.fromMaterializer(
        (mat, attr) -> {
          GoogleSettings settings = GoogleAttributes.resolveSettings(mat, attr);
          return Source.empty();
        });
    // #accessing-settings
  }

  void customSettings() {
    // #custom-settings
    stream.addAttributes(GoogleAttributes.settingsPath("my-app.custom-google-config"));

    GoogleSettings defaultSettings = GoogleSettings.create(system);
    GoogleSettings customSettings = defaultSettings.withProjectId("my-other-project");
    stream.addAttributes(GoogleAttributes.settings(customSettings));
    // #custom-settings
  }
}

File: google-common/src/test/scala/docs/scaladsl/GoogleCommonDoc.scala
/*
 * Copyright (C) since 2016 Lightbend Inc. <https://www.lightbend.com>
 */

package docs.scaladsl

import akka.actor.ActorSystem
import akka.stream.Graph
import akka.stream.alpakka.google.{GoogleAttributes, GoogleSettings}
import akka.stream.scaladsl.Source

import scala.annotation.nowarn

@nowarn
class GoogleCommonDoc {

  implicit val system: ActorSystem = ???
  val stream: Graph[Nothing, Nothing] = ???

  { //#accessing-settings
    val defaultSettings = GoogleSettings()
    val customSettings = GoogleSettings("my-app.custom-google-config")
    Source.fromMaterializer { (mat, attr) =>
      val settings: GoogleSettings = GoogleAttributes.resolveSettings(mat, attr)
      Source.empty
    }
    //#accessing-settings
  }

  {
    //#custom-settings
    stream.addAttributes(GoogleAttributes.settingsPath("my-app.custom-google-config"))

    val defaultSettings = GoogleSettings()
    val customSettings = defaultSettings.withProjectId("my-other-project")
    stream.addAttributes(GoogleAttributes.settings(customSettings))
    //#custom-settings
  }

}
